### PR TITLE
Changes to support social metrics API optimizations

### PR DESCRIPTION
**Pull request by ziafazal**
This PR has changes to support social metrics API optimizations. This YONK stories has details of the API optimizations need.
https://openedx.atlassian.net/browse/YONK-581
https://openedx.atlassian.net/browse/YONK-582
https://openedx.atlassian.net/browse/YONK-584

@ayesha-baig would you please review?

### CODE REVIEW

File: social_engagement/models.py [hunk 1]
```diff
@@ -103,14 +109,19 @@ def generate_leaderboard(cls, course_key, count=3, exclude_users=None, org_ids=N
...
-            'user__username',
-            'user__profile__title',
-            'user__profile__avatar_url',
-            'score')\
-            .order_by('-score', 'modified')[:count]
-        return queryset
+
+        aggregates = queryset.aggregate(Avg('score'))
+        course_avg = aggregates['score__avg']
+        if count:
```
Comment (ayesha-baig) [inline social_engagement/models.py]: Do we really need an if condition here, when the default value of count is set to 3?

Comment (ayesha-baig) [reply social_engagement/models.py]: @ziafazal

**Comment by ayesha-baig** [comment]
Just a single query, rest of the changes are fine.

File: social_engagement/models.py (hunk 1 above)
Comment (ziafazal) [reply social_engagement/models.py]: this condition is for use case when we have `count > 0` in case of `count = 0` we don't want to execute that block.

**Review by ayesha-baig — APPROVED** [review]

**Pull request merged**